src/studio/src/designer/backend/Controllers/SessionController.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Altinn.Studio.Designer.Configuration;
+
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+
+namespace Altinn.Studio.Designer.Controllers
+{
+    /// <summary>
+    /// Controller that exposes endpoints for hangling everything related to the user's session in Altinn Studio.
+    /// </summary>
+    [ApiController]
+    [Route("/designer/api/v1/session")]
+    [AutoValidateAntiforgeryToken]
+    public class SessionController : ControllerBase
+    {
+        private readonly GeneralSettings _settings;
+        private readonly int _sessingExtensionInMinutes = 30;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SessionController"/> class.
+        /// </summary>
+        /// <param name="settings">The general settings.</param>
+        public SessionController(IOptions<GeneralSettings> settings)
+        {
+            _settings = settings.Value;
+        }
+
+        /// <summary>
+        /// Gets the value of the session timeout cookie.
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("remaining")]
+
+        public int GetRemainingSessionTime()
+        {
+            HttpContext.Request.Cookies.TryGetValue(_settings.SessionTimeoutCookieName, out string remainingString);
+
+            if (string.IsNullOrEmpty(remainingString) || !DateTime.TryParse(remainingString, out DateTime timeout))
+            {
+                return -1;
+            }
+
+            return (int)Math.Floor((timeout - DateTime.UtcNow).TotalMinutes);
+        }
+
+        /// <summary>
+        /// Extends the current session with the defined extension length.
+        /// </summary>
+        /// <returns></returns>
+        [HttpPut]
+        [Route("keepAlive")]
+        public async Task<ActionResult> KeepAlive()
+        {
+            AuthenticateResult ar = await HttpContext.AuthenticateAsync();
```

Comment:
Could this be an attribute on the method instead? I assume this is a form of authentication check?

Reply:
No, because the failing authentication would be caught by the middleware and we would get a redirect rather than a 401 response.

Reply:
The question then is if we should consider adding custom middleware to change that behavior. This is an issue with a mix of Views and pure API endpoints.

Reply:
How much work would that be to implement, do you think? Would it make more sense to prioritize re-writing the views into react?

Reply:
There is actually already an anonymous function defined in `AuthenticationConfiguration.ConfigureAuthentication()` on the `OnRedirectToLogin` property of `CookieAuthenticationEvents`. Maybe we can expand the logic to conditionally return 401 in this specific case?

Reply:
@acn-sbuad Could we avoid using uppercase in the route?